src/octonode/client.coffee

```diff
@@ -142,37 +144,106 @@ class Client
         body = JSON.parse(body || '{}')
       catch err
         return callback(err)
-
+    debugger
     return callback(new HttpError(body.message, res.statusCode, res.headers, body)) if body.message and res.statusCode in [400, 401, 403, 404, 410, 422]
     callback null, res.statusCode, body, res.headers
 
+  errorHandlePromise: (response, cb) ->
+    debugger
+    {statusCode, body, headers} = response
+
+    fiveError = new HttpError('Error ' + statusCode, statusCode, headers)
+    fourError = new HttpError(body.message, statusCode, headers, body)
+
+    return cb(fiveError) if Math.floor(statusCode/100) is 5
+    return cb(fourError) if body.message and Math.floor(statusCode/100) is 4
+
+    return cb(null, statusCode, body, headers)
+
   # Github api GET request
-  get: (path, params..., callback) ->
-    @request @requestOptions(
+
+  get: (path, params..., cb) =>
+
+    cb = if cb then cb else () -> arguments
+
+    options = @requestOptions(
       uri: @buildUrl path, params...
       method: 'GET'
-    ), (err, res, body) =>
-      return callback(err) if err
-      @errorHandle res, body, callback
-
-  # Github api GET request
-  getNoFollow: (path, params..., callback) ->
-    @request @requestOptions(
+      followRedirect: true
+      json: true
+      simple: false
+      resolveWithFullResponse: true
+    )
+
+    return @rp options
+      .then((response) => @errorHandlePromise(response, cb))
+      .catch((err) -> cb(err))
```

Comment:
Same issue as above where promises are not handled correctly.  This would need to reject the promise if there is an error or resolve/return if it's successful.  If no `cb` is passed in this code and there is a hard error, it will `catch`, then call you fall through callback.  When you return in a `catch`, the error doesn't propagate.  As in, the next `then` function called on this promise will get triggered, not the next `catch`.  The person consuming this and using it as a promise lib would expect it to trigger the `catch`.    Would be as simple as just removing this line